mrmustard/physics/gaussian_integrals.py

```diff
@@ -131,20 +132,19 @@ def complex_gaussian_integral(
     M = math.gather(math.gather(A, idx, axis=-1), idx, axis=-2) + X * measure
     bM = math.gather(b, idx, axis=-1)
 
-    not_idx = tuple(i for i in range(A.shape[-1]) if i not in idx)
+    c_post = (
+        c * math.sqrt((-1) ** n / math.det(M)) * math.exp(-0.5 * math.sum(bM * math.solve(M, bM)))
+    )
+
     if math.asnumpy(not_idx).shape != (0,):
         D = math.gather(math.gather(A, idx, axis=-1), not_idx, axis=-2)
         R = math.gather(math.gather(A, not_idx, axis=-1), not_idx, axis=-2)
         bR = math.gather(b, not_idx, axis=-1)
         A_post = R - math.matmul(D, math.inv(M), math.transpose(D))
         b_post = bR - math.matvec(D, math.solve(M, bM))
     else:
-        A_post = math.astensor([])
-        b_post = math.astensor([])
-
-    c_post = (
-        c * math.sqrt((-1) ** n / math.det(M)) * math.exp(-0.5 * math.sum(bM * math.solve(M, bM)))
-    )
+        A_post = math.zeros((0, 0), dtype=A.dtype)
```

Comment:
Why it is not empty but instead with (0,0)?

Reply:
so that it sill has the right len(shape)